consensus/consensus_service.go

```diff
@@ -273,9 +274,7 @@ func (consensus *Consensus) checkViewID(msg *FBFTMessage) error {
 
 // SetBlockNum sets the blockNum in consensus object, called at node bootstrap
 func (consensus *Consensus) SetBlockNum(blockNum uint64) {
```

Comment:
Do you think there is a race condition at `consensus/consensus_v2.go:263`.

Reply:
Agree. there could be race condition in all the reading.  Our original code wasn't protecting all the read operation at all.  I'd do it later.  This PR is getting too big now.

Reply:
updated with atomic.AddUint64